### PR TITLE
Add sun_gtil2 component

### PR DESCRIPTION
**Pull request by Mat931**
## Description:
Documentation for the sun_gtil2 component

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4958

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sun_gtil2.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+need to set the ``baud_rate`` to 9600.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    uart:
+      - id: control_to_display
+        rx_pin: GPIO16
+        baud_rate: 9600
+
```
Comment (jesserockz) [inline components/sun_gtil2.rst]: Dependent components are not to be included in examples
```suggestion
```

File: components/sun_gtil2.rst [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        state:
+          id: gtil_state
+          name: "State"
+        serial_number:
+          id: gtil_serial_number
+          internal: True
+
+The inverter will report new measurements approximately every 1.3 seconds.
+
+Configuration variables:
```
Comment (jesserockz) [inline components/sun_gtil2.rst:92]: Please see `pylontech.rst` for an example on splitting the configuration variables per platform type.

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/sun_gtil2.rst (hunk 1 above)
Comment (Mat931) [reply components/sun_gtil2.rst]: Should I put that somewhere else or just leave it out completely?

Comment (jesserockz) [reply components/sun_gtil2.rst]: Just leave it out, its up to the user to read the docs on uart and configure it according to their physical board and pins.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**